include/seastar/core/preempt.hh

```diff
@@ -50,9 +50,10 @@ inline bool need_preempt() {
     // Possible optimization: read head and tail in a single 64-bit load,
     // and find a funky way to compare the two 32-bit halves.
     return __builtin_expect(head != tail, false);
-#else
-    return true;
-#endif
 }
+#define need_preempt() __builtin_expect(::seastar::_need_preempt(), false)
```

Comment:
:+1: 

not sure of upstream would want to upper-case this function as it's a macro now.